### PR TITLE
fix duplicate key exception in mssql

### PR DESCRIPTION
**Pull request by cgardens**
## What
* Removes temporary hack to unblock use of mssql source. (https://github.com/airbytehq/airbyte/pull/982)
* Enforce that the MSSQL does not return duplicate columns for any table.

## Checklist
- [x] *Write a test*

### CODE REVIEW

File: airbyte-protocol/models/src/main/java/io/airbyte/protocol/models/Field.java [hunk 1]
```diff
@@ -55,4 +57,21 @@ public String getTypeAsJsonSchemaString() {
     return type.name().toLowerCase();
   }
 
+  @Override
+  public boolean equals(Object o) {
```
Comment (sherifnada) [inline airbyte-protocol/models/src/main/java/io/airbyte/protocol/models/Field.java:61]: can we use equals builder for easier extensibility?

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java [hunk 2]
```diff
@@ -96,7 +96,9 @@ protected String getCurrentDatabaseName(DSLContext ctx) {
               final String dataType = r.get("data_type", String.class);
               final JsonSchemaPrimitive jsonType = getType(dataType);
               return Field.of(columnName, jsonType);
-            }).collect(Collectors.toList());
+            })
+            .distinct() // this query can return duplicate columns if a column is used in multiple indexes or as a primary key.
+            .collect(Collectors.toList());
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java:101]: would it make sense to also collect to a set, or does order matter?

**Review by sherifnada — APPROVED** [review]

File: airbyte-protocol/models/src/main/java/io/airbyte/protocol/models/Field.java (hunk 1 above)
Comment (cgardens) [reply airbyte-protocol/models/src/main/java/io/airbyte/protocol/models/Field.java:61]: this is still using the intellij template. happy to use a different template if you have one you want to share.

File: airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java (hunk 2 above)
Comment (cgardens) [reply airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java:101]: an interesting question. i think we don't take advantage of it now, but i we may want to have the order in this these lists be the actual order of columns in the table created in the destination.

**Review by jrhizor — APPROVED** [review]

**Pull request merged**